docs/current/sdk/go/guides/205271-replace-dockerfile.md

```diff
@@ -0,0 +1,74 @@
+---
+slug: /205271/replace-dockerfile
+displayed_sidebar: 'current'
+---
+
+# Replace a Dockerfile with Go
+
+## Introduction
+
+This guide explains how to use the Dagger Go SDK to perform all the same operations that you would typically perform with a Dockerfile, except using Go. You will learn how to:
+
+- Create a Dagger client in Go
+- Use Dagger to:
+  - Configure a container with all required dependencies and environment variables
+  - Download and build the application source code in the container
+  - Set the container entrypoint
+  - Publish the built container image to Docker Hub
+
+## Requirements
+
+This guide assumes that:
+
+- You have a Go development environment with Go 1.15 or later. If not, [download and install Go](https://go.dev/doc/install).
+- You have Docker installed and running on the host system. If not, [install Docker](https://docs.docker.com/engine/install/).
+- You have a Go module with the Dagger Go SDK installed. If not, [install the Dagger Go SDK](../371491-install.md).
+
+## Step 1: Understand the source Dockerfile
+
+To illustrate the process, this guide replicates the build process for the popular open source [Memcached caching system](https://www.memcached.org/) using Dagger. It uses the Dockerfile for the [official Docker Hub Memcached image](https://github.com/docker-library/memcached).
+
+Begin by reviewing the [Dockerfile](https://github.com/docker-library/memcached/blob/1e3f84629bb2ab9975235401c716c1e00563fa82/alpine/Dockerfile) to understand how it works. This Dockerfile is current at the time of writing and is available under the BSD 3-Clause License.
```

Comment:
Copyright/legal question here. Is it OK to copy the source Dockerfile into our tutorial? I've not done it right now out of caution, instead linking to the source commit ID (which is fine except that the user needs an extra click to see it).

Reply:
Update: I think it's better to link to it, so I'll leave this as is unless there are strong feelings that it should change (eg. to make the tutorial easier to follow)